### PR TITLE
Support Spring Controllers throwing ResponseStatusException

### LINKED ISSUE #9055: Lack of Spring `org.springframework.web.server.ResponseStatusException`
**Describe the bug**
Spring proposes the `ResponseStatusException` to handle java exception in an HTTP-friendly manner.

When migrating to Quarkus, this class is not provided by the `quarkus-spring-web` compatibility extension.

**Expected behavior**
Being able to reuse a spring controller with the minimum of change 

**Actual behavior**
The code does not compile until replacing Spring `ResponseStatusException` by JAX-RS`WebApplicationException`

**To Reproduce**
1. The code is here: https://github.com/danielpetisme/quarkus-universal-todo/tree/spring-response-issue
2. Start up the infrastructure `docker-compose -f infrastructure/docker-compose.yaml up`
3. Go to the `quarkus-spring` folder
3. `./mvnw compile quarkus:dev` the compilation should failed pointing https://github.com/danielpetisme/quarkus-universal-todo/blob/spring-response-issue/spring/src/main/java/me/escoffier/TodoResource.java

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
(Add any other context about the problem here.)

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #9055 

~~Depends on: https://github.com/quarkusio/quarkus-spring-api/pull/13~~

### CODE REVIEW

File: extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseStatusExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.web.server.ResponseStatusException;
+
+public class ResponseStatusExceptionMapper implements ExceptionMapper<ResponseStatusException> {
+
+    @Override
+    public Response toResponse(ResponseStatusException exception) {
+        Response.ResponseBuilder responseBuilder = new ResponseBuilderImpl().status(exception.getStatus().value());
+        addHeaders(responseBuilder, exception.getResponseHeaders());
+        return responseBuilder.entity(exception.getMessage())
+                .type(MediaType.TEXT_PLAIN).build();
```
Comment (gsmet) [inline extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseStatusExceptionMapper.java:21]: Does Spring always send `text/plain`? I would expect it to be clever than that.

Comment (geoand) [reply extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseStatusExceptionMapper.java:21]: Fair point, but we have been using `text/plain` for exceptions annotated with `@ResponseStatus` (which is very similar to this) and we haven't had anyone complain so far.

We can certainly reexamine this on a grander scale if it becomes a problem.

**Comment by geoand** [comment]
Marking as ready for review since the new dependency has made its way to Central

**Comment by geoand** [comment]
@gsmet can we merge this one?

**Review by gsmet — APPROVED** [review]

**Pull request merged**